### PR TITLE
Add interop matrix 1.47.1 for C++, Python, Ruby and PHP

### PR DESCRIPTION
**Pull request by jtattermusch**
- 1.47.x release is missing in the interop matrix
- adjust the interop matrix docs to stop including C# for new releases (as C# has been removed from the master branch).

### CODE REVIEW

**Comment by jtattermusch** [comment]
Test failures are unrelated. This is ready to merge once approved.

**Review by gnossen — APPROVED** [review]

**Pull request merged**
